model2vec/distill/tokenizer.py

```diff
@@ -109,7 +114,19 @@ def add_tokens(tokenizer: Tokenizer, tokens_to_add: list[str]) -> Tokenizer:
                 wordpiece_vocab[token] = len(wordpiece_vocab)
 
     elif model == "Unigram":
-        raise ValueError("Adding tokens to a unigram tokenizer is not supported.")
+        pre_tokenize = tokenizer.backend_tokenizer.pre_tokenizer.pre_tokenize_str
+        pre_tokenizer = data["pre_tokenizer"]
+        if pre_tokenizer['type'] != 'Metaspace':
+             raise ValueError("Only Metaspace pre-tokenizer is supported for Unigram tokenizers.")
+        elif pre_tokenizer['prepend_scheme'] != 'always':
+            raise ValueError("Only prepend_scheme 'always' is supported for Unigram tokenizers with Metaspace pre-tokenizer.")
+        else:
+            unigram_vocab: list[tuple[str, float]] = data["model"]["vocab"]
+            vocab = set(item[0] for item in unigram_vocab)
+            for token in tokens_to_add:
+                if token not in vocab:
+                    for (pre_tokenized_token, _) in pre_tokenize(token):
```

Comment:
This is unfortunately not correct. Pretokenization can split tokens into multiple individual tokens, which can be in vocab separately. So for example, if you pre-tokenize "water-bender", this could get pretokenized into `["water", "-", "bender"]`. While "water-bender" isn't in vocabulary, the individual words might be. So the correct solution is to:

1. pre-tokenize
2. if len(pre-tokenized) > 1: discard
3. only check for vocabulary after.

Reply:
I think these type of tokens should be removed before starting distillation for all tokenizers/pretokenizers
Do you agree?